File: test/fixtures/negatives/streamable-http-server.ts
// A correctly migrated Streamable HTTP MCP server. It legitimately mentions
// text/event-stream — Streamable HTTP frames POST responses as SSE — but it
// serves ONE endpoint and never writes an `event: endpoint` frame, so the
// HTTP+SSE transport rule must NOT fire (nor any other rule).

import { StreamableHTTPServerTransport } from '@modelcontextprotocol/sdk/server/streamableHttp.js';
import express from 'express';

const app = express();

app.post('/mcp', express.json(), async (req, res) => {
  res.setHeader('Accept', 'application/json, text/event-stream');
  const transport = new StreamableHTTPServerTransport({
    sessionIdGenerator: undefined,
  });
  await handle(transport, req, res);
});

declare function handle(t: unknown, req: unknown, res: unknown): Promise<void>;

export { app };

File: test/fixtures/negatives/streamable_http_server.py
# A correctly migrated FastMCP server on Streamable HTTP. It accepts
# text/event-stream responses (Streamable HTTP frames POST responses as SSE)
# but runs a single-endpoint transport — the HTTP+SSE rule must NOT fire.

from mcp.server.fastmcp import FastMCP

mcp = FastMCP("weather")

ACCEPT = "application/json, text/event-stream"


@mcp.tool()
async def get_forecast(city: str) -> str:
    return f"Forecast for {city}"


if __name__ == "__main__":
    mcp.run(transport="streamable-http")
